### ISSUE TITLE
Mysql_user Fails if Sql-Mode is ANSI

### ISSUE THREAD

**Issue by bbatsche** [ansible/ansible-modules-core#1870]
**Issue Type:** Bug Report

**Ansible Version:**

```
ansible 1.9.2
  configured module search path = None
```

**Environment:**
- Host: OS X 10.10.4
- Guest: Ubuntu 14.04

**Summary:**

If the `sql-mode` for MySQL includes `ANSI`, using `mysql_user` to assign privileges for a given database will work the first time. However, addition calls to `mysql_user` will fail with the error

```
1141, "There is no such grant defined for user 'username' on host 'localhost'"
```

**Steps To Reproduce:**
1. Add `sql-mode = ANSI` to `my.cnf` in the managed environment
2. Run the following task at least twice
   
   ``` yml
   - mysql_user:
     name:     new_username
     password: terrible_password
     priv:     db_name.*:ALL
     state:    present
   ```

**Expected Results:** First run through should be `changed`, subsequent runs should be `ok`.

**Actual Results:** First run through is `changed`, subsequent runs `fail`.

**Additional Notes:**

My hypothesis is that the `mysql_user` module is failing to correctly parse the output from `SHOW GRANTS` in this case. This is causing the check for whether grants need to be modified to receive a false positive. By default, `SHOW GRANTS` quotes table names with backticks (```) like the following:

```
GRANT ALL PRIVILEGES ON `db_name`.* TO 'new_username'@'localhost'
```

When `sql-mode` includes `ANSI`, database names are quoted with double quotes (`"`):

```
GRANT ALL PRIVILEGES ON "db_name".* TO 'new_username'@'localhost'
```

**Comment by inadarei** [comment]
I got the same exact error. First run fine, subsequent ones fail for no good reason :(

**Comment by bikramwp** [comment]
The query generated for the grant is:

```
GRANT SELECT,INSERT,UPDATE,DELETE,CREATE,DROP,INDEX,ALTER,LOCK TABLES ON `somedatabase`.* TO %s@%s
```

and for revoke the query generated is:

```
REVOKE SELECT,INSERT,UPDATE,DELETE,CREATE,DROP,INDEX,ALTER,LOCK TABLES ON `"somedatabase"`.* FROM %s@%s
```

Note the `somedatabase` vs `"somedatabase"`.

These lines cause this particular issue:

https://github.com/ansible/ansible-modules-core/commit/10ebcccedb542c7e1c499e77a1f53da98d373bc3#diff-b4dbfbaa641f4d515d58a88ce227a981R301

https://github.com/ansible/ansible-modules-core/commit/10ebcccedb542c7e1c499e77a1f53da98d373bc3#diff-b4dbfbaa641f4d515d58a88ce227a981R305

**Comment by kristous** [comment]
Are there any news on this?

Addition: Same issue but I haven't set a sql mode. If query the sql mode it says NO_ENGINE_SUBSTITUTION

**Comment by Jmainguy** [comment]
@kristous @bbatsche @inadarei @bikramwp give #2764 a shot and see if it solves the issue for you, seems to work for me.

**Comment by Jmainguy** [comment]
@bcoca I think this can be closed.

**Comment by bcoca** [comment]
closing via merged #2764
